src/posix/platform/hdlc_interface.cpp

```diff
@@ -167,6 +171,8 @@ otError HdlcInterface::Init(const RadioUrl &aRadioUrl)
         ExitNow(error = OT_ERROR_INVALID_ARGS);
     }
 
+    mRadioUrl = aRadioUrl;
```

Comment:
Assigning a `RadioUrl` isn't safe. The pointers in `otUrl` will point to the buffer in the old one, and become dangling pointers.

We need to reinitialize the `RadioUrl` here, and maybe reconsider the design of `RadioUrl` to apply RAII.

Reply:
Fixed by overriding assignment operator.

Reply:
I needed to changed my previous solution because of last changes in: #6543 
Currently changing the reference object RadioUrl to static we prevent from having dangling pointers.

Reply:
You can run `script/make-pretty clang-format` to fix the pretty issue. @lmaciejonczyk

Reply:
Thanks